packaging/greenboot/microshift-running-check.sh

```diff
@@ -25,19 +25,61 @@ function forced_termination() {
     exit 1
 }
 
-# Check preconditions for existence of topolvm deployment.
+# Check preconditions for existence of lvms deployment.
 # Adapted from MicroShift code.
 #
 # args: None
-# return: 0 if topolvm readiness should be checked, 1 otherwise
-function topolvmShouldBeDeployed() {
+# return: 0 if lvms readiness should be checked, 1 otherwise
+function lvmsShouldBeDeployed() {
     if ! hash vgs 2>/dev/null; then
         return 1
     fi
-
     if [ -f /etc/microshift/lvmd.yaml ]; then
         return 0
     fi
+    if [ -f /etc/microshift/config.yaml ]; then
+        # The python script checks the config's .storage.driver value. Parsing errors are ignored, these will be caught
+        # by MicroShift during startup, before this greenboot script is run. The script favors verbosity over simplified
+        # code.
+        # args:
+        #   None
+        # return:
+        #   None
+        # exit code:
+        #   0: The lvms components should be checked for. This case occurs for any one of:
+        #       - .storage.driver == one of ["", "lvms"]
+        #       - .storage.driver is unset (indicates default config)
+        #       - .storage is unset (indicates default config)
+        #       - the file cannot be opened (indicates default config)
+        #   1: The lvms components should not be checked for. This case occurs for:
+        #       - .storage.driver == "none"
+        if python -c \
+'import yaml, sys
+try:
+    with open("/etc/microshift/config.yaml") as stream:
+        try:
+            config = yaml.safe_load(stream)
+            if config["storage"]["driver"] == "none":
+                sys.exit(1)
+            elif config["storage"]["driver"] == "lvms":
+                sys.exit(0)
+            elif config["storage"]["driver"] == "":
+                sys.exit(0)
+            else:
+                sys.exit(1)
+        except KeyError as e:
+            sys.exit(0)
+        finally:
+            stream.close()
+except Exception as e:
+    print(f"error opening config: {e}", file=sys.Stderr)
+';
+        then
+            : # noop. config indicates lvms should be deployed, pending the following checks
```

Comment:
Should we use `if ! python...` to avoid this?

Reply:
We can. I opted for verbosity to make it obvious how the py snippet's output is interpreted.